python/lsst/ctrl/bps/wms/htcondor/htcondor_service.py

```diff
@@ -664,8 +687,36 @@ def _handle_job_inputs(generic_workflow: GenericWorkflow, job_name: str, use_sha
     inputs = []
     for gwf_file in generic_workflow.get_job_inputs(job_name, data=True, transfer_only=True):
         _LOG.debug("src_uri=%s", gwf_file.src_uri)
-        if not use_shared or not gwf_file.job_shared:
-            inputs.append(os.path.relpath(gwf_file.src_uri, out_prefix))
+
+        uri = Path(gwf_file.src_uri)
+
+        # Note if use_shared and job_shared, don't need to transfer file.
+
+        if not use_shared:  # Copy file using push to job
+            inputs.append(str(uri.relative_to(out_prefix)))
+        elif not gwf_file.job_shared:  # Jobs require own copy
+
+            # if using shared filesystem, but still need copy in job. Use
+            # HTCondor's curl plugin for a local copy.
+
+            # Execution butler is represented as a directory which the
+            # curl plugin does not handle. Taking advantage of inside
+            # knowledge for temporary fix until have job wrapper that pulls
+            # files within job.
+            if gwf_file.name == "butlerConfig":
+                # The execution butler directory doesn't normally exist
+                # until the submit phase so checking for suffix instead
+                # of using is_dir().
+                if uri.suffix:  # Single file, so just copy.
```

Comment:
A lot of this code would be much simpler if `ButlerURI` were used everywhere.

Reply:
Can you provide an example of how?  The code still has to know whether the URI would represent a file or a directory (which doesn't exist yet).

Reply:
If this was a ButlerURI already that was by definition known to be a directory then it could be asked.  I was talking about the lines below where you create file:// strings.